File: ftp_upload.py
import ftplib
import os
import sys

host = 'ftpupload.net'
user = 'if0_42776222'
password = 'villabeto05'

def ensure_dir(ftp, remote_dir):
    try:
        ftp.cwd(remote_dir)
        ftp.cwd('..')
    except ftplib.error_perm:
        try:
            ftp.mkd(remote_dir)
        except Exception:
            pass

def upload_directory(ftp, local_dir, remote_dir):
    ensure_dir(ftp, remote_dir)
    ftp.cwd(remote_dir)
    for item in os.listdir(local_dir):
        local_path = os.path.join(local_dir, item)
        if os.path.isfile(local_path):
            with open(local_path, 'rb') as f:
                print(f"Uploading {local_path} to {remote_dir}/{item}")
                ftp.storbinary(f'STOR {item}', f)
        elif os.path.isdir(local_path):
            if item == 'documents' and 'public' in local_path and 'uploads' in local_path:
                print(f"Skipping {local_path}")
                continue
            upload_directory(ftp, local_path, item)
    ftp.cwd('..')

try:
    print(f"Connecting to FTP {host}...")
    ftp = ftplib.FTP(host)
    ftp.login(user, password)
    print("Logged in successfully.")
    
    ftp.cwd('/htdocs')
    print("Changed to /htdocs directory.")
    
    base_dir = r"c:\Users\ADMIN\Desktop\Portfolio"
    
    # 1. Upload .htaccess
    with open(os.path.join(base_dir, '.htaccess'), 'rb') as f:
        print("Uploading .htaccess")
        ftp.storbinary('STOR .htaccess', f)
        
    # 2. Upload .env
    env_content = """APP_URL=http://ftpupload.net
DB_HOST=sql107.infinityfree.com
DB_PORT=3306
DB_NAME=if0_42776222_portfolio
DB_USER=if0_42776222
DB_PASSWORD=villabeto05
OWNER_EMAIL=aimievillabeto@gmail.com
OWNER_PASSWORD=@avllbt
"""
    env_path = os.path.join(base_dir, 'prod.env')
    with open(env_path, 'w') as f:
        f.write(env_content)
        
    with open(env_path, 'rb') as f:
        print("Uploading .env")
        ftp.storbinary('STOR .env', f)
        
    # 3. Upload app and public directories
    upload_directory(ftp, os.path.join(base_dir, 'app'), 'app')
    upload_directory(ftp, os.path.join(base_dir, 'public'), 'public')
    
    ftp.quit()
    print("Upload completed successfully.")
except Exception as e:
    print(f"FTP Error: {e}")

File: ftp_upload.php
<?php
$host = 'ftpupload.net';
$user = 'if0_42776222';
$password = 'villabeto05';

$conn = ftp_connect($host);
if (!$conn) {
    die("Could not connect to $host\n");
}

$login = ftp_login($conn, $user, $password);
if (!$login) {
    die("Could not login\n");
}

ftp_pasv($conn, true);
echo "Connected successfully.\n";

if (!ftp_chdir($conn, 'htdocs')) {
    die("Could not change dir to htdocs\n");
}

function upload_dir($conn, $local_dir, $remote_dir) {
    echo "Uploading directory $local_dir to $remote_dir\n";
    @ftp_mkdir($conn, $remote_dir);
    if (!ftp_chdir($conn, $remote_dir)) {
        echo "Failed to cd into $remote_dir\n";
        return;
    }
    
    $files = scandir($local_dir);
    foreach ($files as $file) {
        if ($file === '.' || $file === '..') continue;
        $local_path = $local_dir . '/' . $file;
        
        if (is_dir($local_path)) {
            if ($file === 'documents' && strpos(str_replace('\\', '/', $local_path), 'public/uploads/documents') !== false) {
                echo "Skipping $local_path\n";
                continue;
            }
            upload_dir($conn, $local_path, $file);
        } else {
            echo "Uploading file $local_path as $file\n";
            if (!ftp_put($conn, $file, $local_path, FTP_BINARY)) {
                echo "Failed to upload $local_path\n";
            }
        }
    }
    ftp_cdup($conn);
}

$base_dir = __DIR__;

echo "Uploading .htaccess...\n";
if (!ftp_put($conn, '.htaccess', $base_dir . '/.htaccess', FTP_BINARY)) {
    echo "Failed to upload .htaccess\n";
}

echo "Uploading .env...\n";
if (!ftp_put($conn, '.env', $base_dir . '/prod.env', FTP_BINARY)) {
    echo "Failed to upload .env\n";
}

upload_dir($conn, $base_dir . '/app', 'app');
upload_dir($conn, $base_dir . '/public', 'public');

ftp_close($conn);
echo "Upload completed.\n";
